libkbfs/kbpki_client.go

```diff
@@ -243,3 +243,21 @@ func (k *KBPKIClient) loadUnverifiedKeys(ctx context.Context, uid keybase1.UID)
 	[]keybase1.PublicKey, error) {
 	return k.config.KeybaseService().LoadUnverifiedKeys(ctx, uid)
 }
+
+// GetCurrentUserIfLoggedIn returns the current username and uid from
+// kbpki.GetCurrentUserInfo if the user is logged in or empty ones
+// if they are not logged in.
+func GetCurrentUserIfLoggedIn(ctx context.Context, kbpki KBPKI, isPublic bool) (libkb.NormalizedUsername, keybase1.UID, error) {
```

Comment:
This function does a bit more, since it conditions on `isPublic`. But I think that can be removed. In the call from `md_ops.go` below, we actually want this logic:

```
var uid keybase1.UID
if !handle.IsPublic() {
  _, uid, err = md.config.KBPKI().GetCurrentUserInfo(ctx)
  if err != nil {
    return ...
  }
}
```

So can you

1) add a comment for `decryptMDPrivateData` saying that `uid` is ignored (and can be empty) if the MD is for a public TLF,
2) implement that logic above, and
3) Change this function to not take `isPublic`, and to return just the username? (And maybe rename to `GetCurrentUsernameIfLoggedIn`.